### PR TITLE
Fixes #20: Added Max. Simultaneous Download GUI option 

### LINKED ISSUE #20: Enhancement: Option "maximum parallel downloads"
It would be nice if there would be an option to change the maximum downloads.

Currently it's fixed at 3 downloads at each time.

### PR DESCRIPTION
**Pull request by farseenabdulsalam**
Now user can change how many videos are downloaded in parallel by heading over to Options-->Connnection-->Max. Simultaneous Downloads

### CODE REVIEW

File: youtube_dl_gui/optionsframe.py [hunk 1]
```diff
@@ -607,6 +607,7 @@ class ConnectionTab(TabPanel):
 
     SPINCTRL_SIZE = (60, -1)
 
+    MAXSIMDLS_LABEL = _("Max. Simultaneous Downloads")
```
Comment (MrS0m30n3) [inline youtube_dl_gui/optionsframe.py]: You should change this to "Max Simultaneous Downloads" (Remove the dot from Max)

File: youtube_dl_gui/optionsframe.py [hunk 2]
```diff
@@ -633,6 +636,12 @@ def _set_sizer(self):
 
         vertical_sizer.AddSpacer(self.SIZE_10)
 
+        maxsimdls_sizer = wx.BoxSizer(wx.HORIZONTAL)
```
Comment (MrS0m30n3) [inline youtube_dl_gui/optionsframe.py]: You should add the maxsimdls_sizer to the same HORIZONTAL sizer with the retries_spintctrl this way you keep them side by side and it looks better.

Something like this
```python
top_hor_sizer = wx.BoxSizer(wx.HORIZONTAL)                              
                                                                        
top_hor_sizer.Add(self.retries_text)                                    
top_hor_sizer.AddSpacer(self.SIZE_5)                                    
top_hor_sizer.Add(self.retries_spinctrl)                                
                                                                                
top_hor_sizer.AddSpacer(self.SIZE_20)                                   
                                                                                
top_hor_sizer.Add(self.maxsimdls_text)                                  
top_hor_sizer.AddSpacer(self.SIZE_5)                                    
top_hor_sizer.Add(self.maxsimdls_spinctrl)                              
                                                                               
vertical_sizer.Add(top_hor_sizer)
```

File: youtube_dl_gui/optionsframe.py [hunk 3]
```diff
@@ -633,11 +636,23 @@ def _set_sizer(self):
 
         vertical_sizer.AddSpacer(self.SIZE_10)
 
-        retries_sizer = wx.BoxSizer(wx.HORIZONTAL)
+        top_hor_sizer = wx.BoxSizer(wx.HORIZONTAL)
+	
+	retries_sizer = wx.BoxSizer(wx.HORIZONTAL)
```
Comment (MrS0m30n3) [inline youtube_dl_gui/optionsframe.py]: You should fix the indentation.
Use 4 spaces per indentation level.

File: youtube_dl_gui/optionsframe.py [hunk 4]
```diff
@@ -633,11 +636,23 @@ def _set_sizer(self):
...
         retries_sizer.Add(self.retries_text)
         retries_sizer.AddSpacer(self.SIZE_5)
         retries_sizer.Add(self.retries_spinctrl)
-        vertical_sizer.Add(retries_sizer)
+
+        maxsimdls_sizer = wx.BoxSizer(wx.HORIZONTAL)
```
Comment (MrS0m30n3) [inline youtube_dl_gui/optionsframe.py]: You don't actually need the *retries_sizer* and *maxsimdls_sizer* sizers.
You can pack them all together in the *top_hor_sizer* which results in fewer lines of code.

**Comment by MrS0m30n3** [comment]
Hi @farseenabdulsalam 
Can you make those changes in order to merge your pull request?

**Comment by nodiscc** [comment]
@farseenabdulsalam has removed his fork and patch. (`unknown repository`). This should be closed.

**Pull request closed without merging**